### DISCUSSION TITLE
Use Limited Access Features in Windows app sdk

### DISCUSSION THREAD

**Discussion by winnative** [microsoft/WindowsAppSDK#4796] [Announcements]
### Describe the bug

I've requested a LAF access token for Taskbar pin to Microsoft and I received it. After that I use TryUnlockFeature method in my project. My app is self-contained unpackaged app so I have to import a .rc file to my project (Based on what Microsoft said) that has a simple line:

Identity LimitedAccessFeature {...}

but I don't know how should I do it.

Can anyone help me?
(The used programming language is C#)

### Steps to reproduce the bug

1. Use your access token in your app project. (Using TryUnlockFeature)
2. See result of it that is unavailable. ( Maybe need to .rc file )

### NuGet package version

Windows App SDK 1.6.1: 1.6.240923002

### Packaging type

Unpackaged

### Windows version

Insider Build (xxxxx)

### IDE

Visual Studio 2022

**Comment by codendone** [comment]
I think the real issue/question here is: **How do you add a Win32 resource to a C# app?**

The `.rc` file here needs to be part of the main app .exe. (Packaged apps use the packaged app identity, so no `.rc` file is needed in that case.)

I don't know if there is a better way to do it, but here's a way you can add Win32 resources to a C# app:
1. Create a C++ project as part of your solution just to use its compiled `.rc` file. As a test, I created a "Windows Desktop Application" C++ project, which comes with a default `.rc` file.
2. Edit the `.rc` file to add the `Identity LimitedAccessFeature {...}` line at the bottom.
3. Build that C++ project and find the path to the compiled `.res` file.
4. Edit the properties of the .csproj, scroll to Application -> Win32 Resources section, change the "Resources" dropdown to "Resource file", and fill in the "Resource file" textbox to the path of the `.res` file (either absolute or relative path is fine).
5. Build and run the C# project.

The `TryUnlockFeature` call should now pick up the Identity specified in the `.rc` file.

Note: You may need to do additional work to add or remove things from the `.rc` file to get it the way you want. Setting Win32 resources turns off default resources which C# would have otherwise included, such as file description and version info. I believe it also removes use of  app.manifest by default, which may need extra work to reconnect that via the `.rc` file.

**Comment by winnative** [comment]
@codendone , thanks for your help.
